### DISCUSSION TITLE
Can I still use pyinstaller -D xxxx.spec in version 5.2?

### DISCUSSION THREAD

**Discussion by cyanghsieh** [pyinstaller/pyinstaller#6967] [PyInstaller] [answered]
When I used version 4.3, I tried `pyi-makespec Main.py` and added my essential folder files into `datas` section in the specfile. e.g.

```
# Main.spec
a = Analysis(['Main.py'],
             pathex=['\\mycode'],
             binaries=[],
             datas=[('\\mycode\\confg','confg')],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
```
then use `pyinstaller -D Main.spec` to build the exe file. This procedure works perfect when using version 4.3.

Now I switch to version 5.2, the same procedure pops out an error:
```
77 INFO: PyInstaller: 5.2
78 INFO: Python: 3.8.10
89 INFO: Platform: Windows-10-10.0.19044-SP0
option(s) not allowed:
  --onedir/--onefile
makespec options not valid when a .spec file is given
```

Is this command `pyinstaller -D xxxx.spec` not available anymore in the future?

**Comment by rokm** [comment, marked as answer]
Switches like `-D` have effect only during initial .spec file generation (e.g., when you build from .py file). If you build from the .spec file, they have no effect anymore (because the contents of the .spec file determine what kind of build will be performed). 

So the behavior was changed in 5.0 ([#6660](https://github.com/pyinstaller/pyinstaller/issues/6660)) to raise an error in order to make this more obvious to users.
